src/test/java/htsjdk/samtools/CRAMAllEncodingStrategiesTest.java

```diff
@@ -21,32 +21,72 @@ public class CRAMAllEncodingStrategiesTest extends HtsjdkTest {
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest_3_containers_aligned_to_pos_1.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false }
         };
     }
 
-    @Test(dataProvider = "roundTripTestFiles")
-    public final void testRoundTripDefaultEncodingStrategy(final File sourceFile, final File referenceFile) throws IOException {
+    @Test(dataProvider = "defaultStrategyRoundTripTestFiles")
+    public final void testRoundTripDefaultEncodingStrategy(
+            final File sourceFile,
+            final File referenceFile,
+            final boolean lenientEquality,
+            final boolean emitDetail) throws IOException {
         final CRAMEncodingStrategy testStrategy = new CRAMEncodingStrategy();
         final File tempOutCRAM = File.createTempFile("testRoundTrip", ".cram");
         tempOutCRAM.deleteOnExit();
         CRAMTestUtils.writeToCRAMWithEncodingStrategy(testStrategy, sourceFile, tempOutCRAM, referenceFile);
-        assertRoundTripFidelity(sourceFile, tempOutCRAM, referenceFile, false);
-        assertRoundtripFidelityWithSamtools(tempOutCRAM, referenceFile);
+        assertRoundTripFidelity(sourceFile, tempOutCRAM, referenceFile, lenientEquality, emitDetail);
+        assertRoundtripFidelityWithSamtools(tempOutCRAM, referenceFile, lenientEquality, emitDetail);
+    }
+
+    @DataProvider(name="encodingStrategiesTestFiles")
+    public Object[][] encodingStrategiesTestFiles() {
+        return new Object[][] {
+                { new File(TEST_DATA_DIR, "NA12878.20.21.1-100.100-SeqsPerSlice.500-unMapped.cram"),
+                        new File(TEST_DATA_DIR, "human_g1k_v37.20.21.1-100.fasta"), false, false },
+        };
     }
 
-    @Test(dataProvider = "roundTripTestFiles")
-    public final void testAllEncodingStrategyCombinations(final File cramSourceFile, final File referenceFile) throws IOException {
+    @Test(dataProvider = "encodingStrategiesTestFiles")
+    public final void testAllEncodingStrategyCombinations(
+            final File cramSourceFile,
+            final File referenceFile,
+            final boolean lenientEquality,
+            final boolean emitDetail) throws IOException {
         for (final Tuple<String, CRAMEncodingStrategy> testStrategy : getAllEncodingStrategies()) {
             final File tempOutCRAM = File.createTempFile("allEncodingStrategyCombinations", ".cram");
             tempOutCRAM.deleteOnExit();
             CRAMTestUtils.writeToCRAMWithEncodingStrategy(testStrategy.b, cramSourceFile, tempOutCRAM, referenceFile);
-            assertRoundTripFidelity(cramSourceFile, tempOutCRAM, referenceFile, false);
-            assertRoundtripFidelityWithSamtools(tempOutCRAM, referenceFile);
+            assertRoundTripFidelity(cramSourceFile, tempOutCRAM, referenceFile, lenientEquality, emitDetail);
+            assertRoundtripFidelityWithSamtools(tempOutCRAM, referenceFile, lenientEquality, emitDetail);
```

Comment:
Add a comment explaining why we also do a test with samtools

Reply:
Its basically testing interoperability with samtools for all of the different encodings, to make sure both tool sets agree on the data. Added a comment saying its for interop testing.